homeassistant/components/cast/media_player.py

```diff
@@ -479,6 +502,30 @@ def play_media(self, media_type, media_id, **kwargs):
         # We do not want this to be forwarded to a group
         self._chromecast.media_controller.play_media(media_id, media_type)
 
+    def cast_app(self, app_name, data, **kwargs):
+        """Launch an application and start playing media."""
+        try:
+            from importlib import import_module
+
+            extended_quick_play = import_module("castextensions").quick_play
```

Comment:
Make `castextenstions` a dependency of either `pychromecast` or `homeassistant.components.cast` instead.

Reply:
Cast extensions has some large dependencies, such as AndroidViewClient. Isn't it a bad idead to add such a dependency to all cast integration users?

Reply:
OK, that's no good.
Can you remove the cast extensions part from this PR? It would be better suited as a custom component.

Reply:
Sure, I'll remove it.

But I'm not exactly sure how to get similar functionality with a custom component. Should I try to extend the cast-component or something like that?

Reply:
A really simple way would be to fork `cast/media_player.py` and use that as a custom component. It will then override the core component.
Not exactly elegant though.

Not sure how to extend the cast-component otherwise, Ideally you'd want to import from `cast/media_player.py` and only override `play_media`.

Reply:
It appears I would need to completely fork `cast` and replace it with `cast_extended` (or something). Doesn't seem ideal.